SampleUsages/TestScenario.cs

```diff
@@ -0,0 +1,169 @@
+            {
+                AssertInput("InputPath", "FunctionName");
+                var urls = File.ReadAllLines(this.InputPath);
+                test = new AzureHttpTriggerTest(this.FunctionName, this.Eps, this.WarmUpTimeInMinutes, urls);
+                inputCount = urls.Count();
+            }
+            else if (scenarioType.Compare(Platform.Azure, TriggerType.Queue))
+            {
+                AssertInput("FunctionName", "InputObject", "OutputObject");
+                var queueMessages = File.ReadAllLines(this.InputPath);
+                test = new AzureQueueTriggerTest(this.FunctionName, this.Eps, this.WarmUpTimeInMinutes, queueMessages, this.InputObject, this.OutputObject);
+                inputCount = queueMessages.Count();
+            }
+            else
+            {
+                var ex = new ArgumentException(
+                        $"Unknown combination of platform ({this.Platform}) and trigger type ({this.TriggerType}) available Platform values are ({string.Join(",", Enum.GetValues(typeof(Platform)))}) available TriggerType values are ({string.Join(",", Enum.GetValues(typeof(TriggerType)))})");
+                this._logger.LogException(ex);
+                throw ex;
+            }
+            
+            profile = GetLoadProfile(inputCount);
+            test.Logger = _logger;
+            RunScenario(test, profile);
+        }
+
+        private TriggerTestLoadProfile GetLoadProfile(int itemsCount)
+        {
+            TriggerTestLoadProfile profile;
+            AssertInput("Eps");
+
+            switch (this.LoadProfile)
+            {
+                case LoadProfilesType.Linear:
+                    if (this.Repeat)
+                    {
+                        AssertInput("DurationInMinutes");
+                        profile = new LinearLoad(TimeSpan.FromMinutes(this.DurationInMinutes),
+                            this.Eps == 0 ? 1 : this.Eps);
+                    }
+                    else
+                    {
+                        profile = new LinearLoad(itemsCount, this.Eps == 0 ? 1 : this.Eps);
+                    }
+                    break;
+                case LoadProfilesType.LinearRampUp:
+                    profile = new LinearWithRampUp(TimeSpan.FromMinutes(this.DurationInMinutes),
+                        this.Eps == 0 ? 1 : this.Eps);
+                    break;
+                default:
+                    var ex = new ArgumentException(
+                        $"Unknown load profile {this.LoadProfile} available values are ({string.Join(",", Enum.GetValues(typeof (LoadProfilesType)))})");
+                    this._logger.LogException(ex);
+                    throw ex;
+            }
+
+            return profile;
+        }
+
+        internal void RunScenario(FunctionTest functionTest, TriggerTestLoadProfile profile)
+        {
+            var perfResult = functionTest.RunAsync(profile).GetAwaiter().GetResult();
```

Comment:
I don't quite get how ConfigureAwait(continueOnCapturedContext: false) would help here. Accessing Result directly will swallow exceptions. A preferable way to use async method synchronously is to do it that way or Task.WaitAll() and handle everything manually.
It is wrapped with try{} catch(){} before it's used.